module/observations/helper.js

```diff
@@ -1048,4 +1052,179 @@ module.exports = class ObservationsHelper {
+
+                if(!observationData || observationData == "") {
+                    return resolve({
+                        message: messageConstants.apiResponses.OBSERVATION_NOT_FOUND,
+                        result: {}
+                    });
+                }
+
+                let appDetails = await kendraService.getAppDetails(appName);
+                
+                if(appDetails.result === false){
+                    throw new Error(messageConstants.apiResponses.APP_NOT_FOUND);
+                }
+                
+                link = process.env.OBSERVATION_SHARE_URL_ENDPOINT+observationData[0].link;
+                return resolve({
+                    message: messageConstants.apiResponses.OBSERVATION_LINK_FETCHED,
+                    result: {"link" :link}
+                });
+               
+                
+            }
+            catch (error) {
+                return reject(error);
+            }
+        })
+    }
+
+    /**
+     * Verfy observation link.
+     * @method
+     * @name verify
+     * @param {Object} data - observation link.
+     * @param {String} requestingUserAuthToken - Requesting user auth token.
+     * @returns {Object} observation data.
+     */
+
+    static verify(
+        data, 
+        requestingUserAuthToken,
+        userId
+        
+    ) {
+        return new Promise(async (resolve, reject) => {
+
+            try {
+                
+                let link = data.link;
+                if(!link || link == "" ) {
+                    return resolve({
+                        message: messageConstants.apiResponses.LINK_REQUIRED_CHECK,
+                        result: []
+                    });
+                }
+
+                let observationSolutionData = await solutionHelper.solutionDocuments({
+                    link: link,
+                    isReusable: false,
+                    type : messageConstants.common.OBSERVATION,
+                    endDate : {$gte: new Date()},
+                    status:messageConstants.common.ACTIVE_STATUS
+                });
+
+                if(!observationSolutionData[0]){
+                    return resolve({
+                        message: messageConstants.apiResponses.LINK_IS_EXPIRED,
+                        result: []
+                    });
+                   
+                }
+
+                if(observationSolutionData[0]){
+
+                    let observationData = await this.observationDocuments({
+                        solutionExternalId : observationSolutionData[0].externalId
+                    });
+
+                    if(observationData){
+                        return resolve({
+                            message: messageConstants.apiResponses.OBSERVATION_LINK_VERIFIED,
+                            result: observationData
+                        });
+                    }
+                    
+                    let userEntities = await database.models.userExtension.findOne({
```

Comment:
@priyanka-TL  Get it from userExtension helper function. Do not use find query direct . Get it from their respective helper.